src/collision.cpp

```diff
@@ -54,6 +54,22 @@ struct NearbyCollisionInfo {
 	aabb3f box;
 };
 
+// Helper functions:
+// Truncate floating point numbers to specified number of decimal places
+// in order to move all the floating point error to one side of the correct value
+static inline f32 truncate(const f32 val, const u16 exp)
+{
+	return std::trunc(val * 10.0f * exp) / (10.0f * exp);
```

Comment:
```suggestion
	return truncf(val * 10.0f * exp) / (10.0f * exp);
```

Reply:
I hear math.h is supposed to be deprecated. How about `std::truncf`?

Reply:
Is it? I thought `std::foo` vs `foo` is just a style choice (assuming no function overloads).

Reply:
yes, looks like in this case they both point to the same place.
Done